torchvision/transforms/functional.py

```diff
@@ -82,6 +82,36 @@ def to_tensor(pic):
         return img
 
 
+def pil_to_tensor(pic, swap_to_channelsfirst=True):
+    """Convert a ``PIL Image`` to a tensor of the same type.
+
+    See ``AsTensor`` for more details.
+
+    Args:
+        pic (PIL Image): Image to be converted to tensor.
+        swap_to_channelsfirst (bool): Boolean indicator to convert to CHW format.
+
+    Returns:
+        Tensor: Converted image.
+    """
+    if not(_is_pil_image(pic)):
+        raise TypeError('pic should be PIL Image. Got {}'.format(type(pic)))
+
+    if accimage is not None and isinstance(pic, accimage.Image):
+        nppic = np.zeros([pic.channels, pic.height, pic.width], dtype=np.float32)
+        pic.copyto(nppic)
+        return torch.as_tensor(nppic)
+
+    # handle PIL Image
+    img = torch.as_tensor(np.asarray(pic))
```

Comment:
This line will still produce the same bug mentioned in #2194. Converting to numpy with `np.asarray(pic)` will keep the PIL image non-writeable. If instead we use `np.array(pic)`, the bug #2194 would not appear. But I believe this is no real fix because `np.array(pic)` copies the data which might be unintended behavior here.

Reply:
I think we will wait until PyTorch fixes this behavior in master, as making a copy would be fairly expensive.

If the warnings are too annoying, an alternative would be to only use `asarray` if the array is writeable.